cyipopt/scipy_interface.py

```diff
@@ -164,26 +211,30 @@ def get_bounds(bounds):
         return lb, ub
 
 
-def get_constraint_bounds(constraints, x0, INF=1e19):
-    if isinstance(constraints, dict):
-        constraints = (constraints, )
+def get_constraint_bounds_and_dimensions(constraints, x0, INF=1e19):
```

Comment:
This is also backwards incompatible. Best to make a new function to get the dimensions.

Reply:
You could add a kwarg that causes the number of returned item to increase in length, that's ok, but still not as clean as it could be.